### PR TITLE
feat(forge): add `forge selectors collision` cmd

### PR DESCRIPTION
**Pull request by lmanini**
## Motivation
Follow up PR for #5012.

## Solution
Adds a subcommand to `forge selectors` to check for methods with colliding selectors between two contracts.
The output format is similar to that of `forge inspect <contract> mi` but it could still be improved if you guys dislike the current format.

To test this I've created a forge repo in the project, added Proxy.sol and Token.sol : [gist](https://gist.github.com/lmanini/ddbb15c190b19fd3c5ff8d8b94a6cd04).
Running `forge selectors collision Token Proxy` prints:
```
"42966c68": ("burn(uint256)", "collate_propagate_storage(bytes16)")
"9ce110d7": ("initializer()", "initializer()")
"06fdde03": ("name()", "name()")
"95d89b41": ("symbol()", "symbol()")
Error: 
At least one collision was found
```

and running `forge selectors collision Proxy Counter` prints:
```
No colliding method selectors between the two contracts.
```

### CODE REVIEW

File: cli/src/cmd/forge/selectors.rs
Comment (Evalir) [inline cli/src/cmd/forge/selectors.rs:154]: this is ok, but maybe we could do this in a more rusty/functional way? something like filtering whichever method map for items contained in the second and collecting this into another vector.

File: cli/src/cmd/forge/selectors.rs
Comment (Evalir) [inline cli/src/cmd/forge/selectors.rs:171]: Hmm, I was thinking, maybe this command doesn't need to error on here? We can color the output with `Paint::red` for saying we found collisions, and then maybe pretty print on a table instead. wdyt @mds1 / @mattsse ?

File: cli/src/cmd/forge/selectors.rs
Comment (Evalir) [inline cli/src/cmd/forge/selectors.rs:37]: is this box necessary?

**Review by Evalir — CHANGES_REQUESTED** [review]
This overall looks good! Just a few comments, mainly bikeshedding on the way we handle the output:

File: cli/src/cmd/forge/selectors.rs
Comment (mds1) [reply cli/src/cmd/forge/selectors.rs:171]: I think a common use of this command is to run it in CI to make sure you don't merge anything with selector collisions, so my suggestion would be to either:
1. Have this command always exit with an error (i.e. `std::process::exit(1)`)` if a collision is found, OR
2. Only exit with an error if a `--check` flag is added

Currently we do both depending on the command,e.g. `forge build --sizes` exits with an error if anything is over the size limit, but `forge fmt` needs a `--check` flag. I can see an argument that the former should also have a `--check` flag, so I'm indifferent here and maybe we start requiring the `--check` flag as a standard?

Also, I'm now realizing this may all be tangential to returning an `Err` from this function 😅

File: cli/src/cmd/forge/selectors.rs
Comment (lmanini) [reply cli/src/cmd/forge/selectors.rs:154]: sounds good! will rewrite this with a functional construct 👍

File: cli/src/cmd/forge/selectors.rs
Comment (lmanini) [reply cli/src/cmd/forge/selectors.rs:171]: I'm more lenient towards adding the `--check` flag so that we can support both of your guys' visions.
Regarding the pretty print on a table, yesterday I was looking at [prettytable-rs](https://github.com/phsym/prettytable-rs) but idk what you guys think about adding it as a dep.

File: cli/src/cmd/forge/selectors.rs
Comment (lmanini) [reply cli/src/cmd/forge/selectors.rs:37]: Rust won't complain without it, but clippy does as without the two elements of the SelectorsSubcommands enum differ quite a bit in size

File: cli/src/cmd/forge/selectors.rs
Comment (mds1) [reply cli/src/cmd/forge/selectors.rs:171]: Other parts of the code use `comfy-table` to pretty-print stuff, that's what `forge build --sizes` uses, I think `forge inspect` too

File: cli/src/cmd/forge/selectors.rs
Comment (Evalir) [inline cli/src/cmd/forge/selectors.rs:178]: Ok so what I'd do for this command is just not return an error at all—printing the table if we find collisions preceded by `x collisions found` is good enough for me. I'd only return an error if the command failed _internally_, as in building the project or trying to extract the selectors

**Review by Evalir — CHANGES_REQUESTED** [review]
this lgtm pending the comment! honestly if we find collisions i would just not return the `Err`, just print the table along with a "Collisions found" on top

**Review by Evalir — APPROVED** [review]
aight! just pushed the last change—looks good! cc @mattsse for final approval/merge

**Review by mattsse — APPROVED** [review]
nice job

lgtm

**Pull request merged**
